src/App.jsx

```diff
@@ -27,46 +27,90 @@ function App() {
     });
   }
 
+  function getOppDirection(dir) {
+    if (dir == DIRECTION.UP) {
+      return DIRECTION.DOWN;
+    } else if (dir == DIRECTION.DOWN) {
+      return DIRECTION.UP;
+    } else if (dir == DIRECTION.RIGHT) {
+      return DIRECTION.LEFT;
+    } else {
+      return DIRECTION.RIGHT;
+    }
+  }
+
   const up = () => {
-    setCurrentDirection(DIRECTION.UP);
+      setCurrentDirection(prevDirection => {
```

Comment:
Don't set the state itself if the entered direction is opposite direction.
if(isOppositeDirection(currentDirection) !== DIRECTION.UP) {
setState()
}

not setState(prev => // then check the logic)